test/conformance/sources_conf_test.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/pkg/apis"
+	"testing"
+)
+func SetupApiServerSource(c *testlib.Client){
+
+}
+func SetupPingSource(c *testlib.Client){
+
+}
+func SetupContainerSource(c *testing..Client){
+
+}
+func TestSourcesStatusConformance(t *testing.T) {
+	SourceStatusTestHelperWithComponentsTestRunner(t,
+		sourcesTestRunner,
+		SetupApiServerSource)
+}
+
+func SourceStatusTestHelperWithComponentsTestRunner(
+	t *testing.T,
+	componentsTestRunner testlib.ComponentsTestRunner,
+	options ...testlib.SetupClientOption,
+) {
+	table := [] struct {
+		name    string
+		feature testlib.Feature
+		// Sources report success via either a Ready condition or a Succeeded condition
+		want    apis.ConditionType
+	} {
+		{
+			"Long lived sources have Ready status condition",
+			testlib.FeatureLongLived,
+			apis.ConditionReady,
+		},
+		{
+			"Batch sources have Succeeded status condition",
+			testlib.FeatureBatch,
+			apis.ConditionSucceeded,
+		},
+
+	}
+	for _, tc := range table {
+		componentsTestRunner.RunTests(t, tc.feature, func(st *testing.T, source metav1.TypeMeta) {
```

Comment:
I suspect here the `RunTests` uses `testlib.Feature` to filter the available sources to run the test to right?

Reply:
yep